libgloss/arc/hl/hl_clock.c

```diff
@@ -0,0 +1,39 @@
+/*********************************************************************
+Copyright  2002-2020 Synopsys Inc.
+*********************************************************************/
+
+#include <errno.h>
+#include <stdint.h>
+#include <time.h>
+
+#include "../arc_timer.h"
+
+#include "hl_toolchain.h"
+#include "hl_api.h"
+
+static clock_t
+hl_clock (void)
+{
+  uint32_t ret;
+  volatile __uncached char *p;
+
+  p = hl_message (HL_SYSCALL_CLOCK, ":i", (uint32_t *) &ret);
+
+  hl_delete ();
+
+  if (p == NULL) {
+    errno = ETIMEDOUT;
+    return -1;
+  }
+
+  return ret;
+}
+
+clock_t
+_clock (void)
```

Comment:
like in the case of arcgv

Reply:
_clock does more than just hl_clock.

Reply:
Just inline. On the other hand hl_clock is only use once here ;)

Reply:
done